token-metadata/program/tests/verify.rs

```diff
@@ -0,0 +1,1928 @@
+            .await;
+
+        // Verify.
+        let payer = context.payer.dirty_clone();
+        let args = VerifyArgs::CreatorV1;
+        let err = da
+            .verify(&mut context, payer, args, None, None, None, None, None)
+            .await
+            .unwrap_err();
+
+        assert_custom_error!(err, MetadataError::CreatorNotFound);
+
+        da.assert_creators_matches_on_chain(&mut context, &creators)
+            .await;
+    }
+
+    #[tokio::test]
+    async fn no_creators_found() {
+        let mut context = program_test().start_with_context().await;
+
+        // Create and mint item.
+        let mut da = DigitalAsset::new();
+        da.create_and_mint_with_creators(
+            &mut context,
+            TokenStandard::ProgrammableNonFungible,
+            None,
+            None,
+            1,
+            None,
+        )
+        .await
+        .unwrap();
+
+        da.assert_creators_matches_on_chain(&mut context, &None)
+            .await;
+
+        // Verify.
+        let payer = context.payer.dirty_clone();
+        let args = VerifyArgs::CreatorV1;
+        let err = da
+            .verify(&mut context, payer, args, None, None, None, None, None)
+            .await
+            .unwrap_err();
+
+        assert_custom_error!(err, MetadataError::NoCreatorsPresentOnMetadata);
+
+        da.assert_creators_matches_on_chain(&mut context, &None)
+            .await;
+    }
+
+    #[tokio::test]
+    async fn pass() {
+        let mut context = program_test().start_with_context().await;
+
+        // Create and mint item.
+        let creator = Keypair::new();
+        airdrop(&mut context, &creator.pubkey(), LAMPORTS_PER_SOL)
+            .await
+            .unwrap();
+
+        let creators = Some(vec![Creator {
+            address: creator.pubkey(),
+            share: 100,
+            verified: false,
+        }]);
+
+        let mut da = DigitalAsset::new();
+        da.create_and_mint_with_creators(
+            &mut context,
+            TokenStandard::ProgrammableNonFungible,
+            None,
+            None,
+            1,
+            creators.clone(),
+        )
+        .await
+        .unwrap();
+
+        da.assert_creators_matches_on_chain(&mut context, &creators)
+            .await;
+
+        // Verify.
+        let args = VerifyArgs::CreatorV1;
+
+        let verified_creators = Some(vec![Creator {
+            address: creator.pubkey(),
+            share: 100,
+            verified: true,
+        }]);
+
+        da.verify(&mut context, creator, args, None, None, None, None, None)
+            .await
+            .unwrap();
+
+        da.assert_creators_matches_on_chain(&mut context, &verified_creators)
+            .await;
+    }
+}
+
+mod verify_collection {
+    use super::*;
+
+    #[tokio::test]
+    async fn delegate_record_wrong_owner() {
+        // See `standard_delegate_fails_collection_created_new_handlers`.
```

Comment:
Is `standard_delegate_fails_collection_created_new_handlers` an unwritten test? I don't see it anywhere.

Reply:
Yeah I renamed it to the shorter `standard_delegate_fails`

Reply:
fixed